packages/material-ui/src/Menu/Menu.js

```diff
@@ -115,37 +168,119 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     }
   });
 
+  const handleSetOpenSubMenuIndex = (value) => {
+    if (value === null) {
+      if (contentAnchorRef.current.parentElement) {
+        contentAnchorRef.current.parentElement.children[openSubMenuIndex].focus();
+      }
+    }
+    setOpenSubMenuIndex(value);
+  };
+
   const items = React.Children.map(children, (child, index) => {
+    if (!React.isValidElement(child)) {
+      return;
+    }
+
+    const { subMenu, onMouseMove: onMouseMoveChildProp } = child.props;
+    const { anchorEl } = other;
+
+    const hasSubMenu = Boolean(subMenu);
+    const parentMenuOpen = Boolean(anchorEl);
+
+    let additionalPropsAdded = false;
+    const additionalProps = {};
+
+    // This is the original purpose of this React.Children.map and is basically unchanged.
     if (index === activeItemIndex) {
-      return React.cloneElement(child, {
+      additionalPropsAdded = true;
+
+      Object.assign(additionalProps, {
         ref: (instance) => {
           contentAnchorRef.current = instance;
           setRef(child.ref, instance);
         },
       });
     }
 
+    // If the current Menu item in this map has a subMenu,
+    // we need the parent Menu to orchestrate its subMenu
+    if (hasSubMenu && parentMenuOpen) {
+      additionalPropsAdded = true;
+
+      const handleArrowRightKeydown = (event) => {
+        if (event.key === 'ArrowRight') {
+          event.preventDefault();
+          setOpenSubMenuIndex(index);
+        }
+      };
+
+      Object.assign(additionalProps, {
```

Comment:
Seems excessive to use `Object.assign`. Is `additionalProps.openSubMenu = ...` not sufficient?

Reply:
I think I just find the syntax of the `Object.assign` calls more readable. There wasn't a specific technical reason. I think this could be refactored to make use of object spread. Object spread is generally a more efficient operation as well, from what I'm reading after a [quick](https://stackoverflow.com/questions/32925460/object-spread-vs-object-assign) [search](https://thecodebarbarian.com/object-assign-vs-object-spread.html). Could also just individually assign each property, as you suggest. I don't have a strong preference.

Reply:
We only use Object.assign (or any built-in for that matter) if we need it. If we don't need it we might as well just set properties. We don't need to spread for one or two properties either.

Reply:
Cool, I'll circle back on this once I get a green CI.

Reply:
Refactored this section and pushed.